examples/c++/msm/example.cu

```diff
@@ -1,59 +1,125 @@
 #include <fstream>
 #include <iostream>
+#include <iomanip>
 
+// select the curve. TODO: can we use curve names instead of IDs?
+#define CURVE_ID 1
 
 // include MSM template
-#include "appUtils/msm/msm.cu"
-// select the curve
-#include "curves/bn254/curve_config.cuh"
-
-using namespace BN254;
+#include "icicle/appUtils/msm/msm.cu"
+using namespace curve_config;
 
 int main(int argc, char* argv[])
 {
   std::cout << "Icicle example: Muli-Scalar Multiplication (MSM)" << std::endl;
-
   std::cout << "Example parameters" << std::endl;
+  int batch_size = 1;
+  std::cout << "Batch size: " << batch_size << std::endl;
   unsigned msm_size = 1048576;
-  std::cout << "msm_size: " << msm_size << std::endl;
-  unsigned bucket_factor = 10;
-  std::cout << "bucket_factor: " << bucket_factor << std::endl;
+  std::cout << "MSM size: " << msm_size << std::endl;
+  int N = batch_size * msm_size;
+  
   std::cout << "Generating random inputs on-host" << std::endl;
-  scalar_t* scalars = new scalar_t[msm_size];
-  affine_t* points = new affine_t[msm_size];
+  scalar_t* scalars = new scalar_t[N];
+  affine_t* points = new affine_t[N];
   projective_t result;
-  for (unsigned i = 0; i < msm_size; i++) {
-    points[i] = (i % msm_size < 10) ? projective_t::to_affine(projective_t::rand_host()) : points[i - 10];
-    scalars[i] = scalar_t::rand_host();
-  }
+  scalar_t::RandHostMany(scalars, N);
+  projective_t::RandHostManyAffine(points, N);
+
+  std::cout << "Configuring MSM to use on-host inputs" << std::endl;
+  // Create a CUDA stream
+  cudaStream_t stream;
+  cudaStreamCreate(&stream);
+  cudaEvent_t start, stop;
+  float time;
+  // Create two events to time the MSM kernel
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+  // Create a device context
+  device_context::DeviceContext ctx = {
+    stream, // stream
+    0,      // device_id
+    0,      // mempool
+  };
+  // Create a MSM configuration
+  msm::MSMConfig config = {
+    ctx,   // DeviceContext
+    0,     // points_size
+    1,     // precompute_factor
+    0,     // c
+    0,     // bitsize
+    10,    // large_bucket_factor
+    batch_size,    
+    false, // are_scalars_on_device
+    false, // are_scalars_montgomery_form
+    false, // are_points_on_device
+    false, // are_points_montgomery_form
+    false,  // are_results_on_device
+    false, // is_big_triangle
+    true,  // is_async
+  };
+
+  std::cout << "Running MSM kernel with on-host inputs" << std::endl;
+  // Record the start event on the stream
+  cudaEventRecord(start, stream);
+  // Execute the MSM kernel
+  msm::MSM<scalar_t, affine_t, projective_t>(scalars, points, msm_size, config, &result);
+  // Record the stop event on the stream
+  cudaEventRecord(stop, stream);
+  // Wait for the stop event to complete
+  cudaEventSynchronize(stop);
+  // Calculate the elapsed time between the start and stop events
+  cudaEventElapsedTime(&time, start, stop);
+  // Destroy the events
+  cudaEventDestroy(start);
```

Comment:
Small note: possible to avoid destroying events here and recreating them on lines 98-99